docs/source/migration-guides.rst

```diff
@@ -0,0 +1,258 @@
+Migration guides
+================
+
+Migration from Waffle 2.2.0 to Waffle 2.3.0
+-------------------------------------------
+
+**Created monorepo**
+
+Packages :code:`waffle-cli`, :code:`waffle-chai`, :code:`waffle-provider`, :code:`waffle-compiler` were migrated to monorepo.
+
+**Created MockProvider class**
+
+We added MockProvider class, it was change the way to create the provider.
+
+*Waffle 2.2.0*
+
+.. code-block:: ts
+
+  await createMockProvider(options);
+
+*Waffle 2.3.0*
+
+.. code-block:: ts
+
+  provider = new MockProvider();
+
+
+:code:`new MockProvider` returned :code:`providers.Web3Provider`, which is the parent class of :code:`MockProvider`.
+
+**Reorganise getWallets() method**
+
+*Waffle 2.2.0*
+
+.. code-block:: ts
+
+  await getWallets(provider);
+
+*Waffle 2.3.0*
+
+.. code-block:: ts
+
+  wallets = new MockProvider().getWallets()
+
+Migration from Waffle 2.3.0 to Waffle 2.4.0
+-------------------------------------------
+
+**Renamed configuration options**
+
+We renamed configuration options to compile contracts.
+
+*Waffle 2.3.0*
+
+.. code-block:: text
+
+  {
+    sourcesPath: "...",
+    targetPath: "...",
+    npmPath: "...",
+    compiler: "...",
+    docker-tag: "...",
+    solcVersion: "...",
+    legacyOutput: "...",
+    allowedPaths: "...",
+    ganacheOptions: "..."
+  }
+
+
+*Waffle 2.4.0*
+
+.. code-block:: text
+
+  {
+    sourceDirectory: "...",
+    outputDirectory: "...",
+    nodeModulesDirectory: "...",
+    compilerType: "...",
+    compilerVersion: "...",
+    compilerAllowedPaths: "..."
+  }
+
+Migration from Waffle 2.5.* to Waffle 3.0.0
+-------------------------------------------
+
+There are some new functionality and some slight refactoring and improved paradigms in Waffle v3.
+
+**Removed deprecated APIs from provider**
+
+In :code:`Waffle 3.0.0` we remove deprecated APIs from provider.
+
+**Swapped arguments for Fixture**
+
+In :code:`Waffle 3.0.0` we swapped arguments for Fixture, because the provider argument is very rarely used compared to wallets.
+So such implantation should be more convenient for users.
+
+*Waffle 2.5.0*
+
+.. code-block:: ts
```

Comment:
Why is the example so verbose? Focus on the core. Define the fixture and call loadFixture on it. No more is needed